drivers/include/bno055.h

```diff
@@ -0,0 +1,456 @@
+ * @ingroup     drivers_saul
+ * @brief       Device driver for the Bosch BNO055 9-axis sensor
+ *
+ * This driver provides @ref drivers_saul capabilities.
+ * @{
+ *
+ * @file
+ * @brief       Device driver interface for the Bosch BNO055 9-axis sensor
+ * This sensor is an 9-DOF IMU with integrated sensorfusion. This makes it
+ * possible to download rotation matrices and orientations directly from the
+ * registers.
+ * This is the reason for the scaling float functions. To use the sensor fusion
+ * data directly in your code it has to be scaled.
+ * A quaternion representation is worthless if the vector length != 1
+ *
+ * Furthermore, it has the following fusion outputs:
+ *   - Quarternion vector.
+ *   - Euler rotation vector.
+ *   - Linear Acceration Vector
+ *   - Gravity Vector
+ *
+ * @note        The current state of the driver only implements basic polling.
+ *
+ * @author      Wouter Horlings <wouter@horlin.gs>
+ */
+
+#ifndef BNO055_H
+#define BNO055_H
+
+#include <stdint.h>
+#include <errno.h>
+
+#include "periph/i2c.h"
+#include "periph/gpio.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+/**
+ * @brief contains 8 unsigned bytes and 4 signed words.
+ */
+typedef union {
+    uint8_t u8bit[8];   /**< Array of 8 unsigned bytes */
+    int16_t i16_bit[4]; /**< Array of 4 signed words */
+} bno055_4bit16_t;
+
+/**
+ * @brief contains 6 unsigned bytes and 3 signed words.
+ */
+typedef union {
+    uint8_t u8bit[6];   /**< Array of 6 unsigned bytes */
+    int16_t i16_bit[3]; /**< Array of 3 signed words */
+} bno055_3bit16_t;
+
+/**
+ * @brief Representing the four components of a quaternion
+ */
+typedef struct {
+    float w;    /**< Real component of quaternion */
+    float x;    /**< First imaginary component of quaternion */
+    float y;    /**< Second imaginary component of quaternion */
+    float z;    /**< Third imaginary component of quaternion */
+} quat_t;
+
+/**
+ * @brief Representing an Euler angle rotation
+ */
+typedef struct {
+    float heading;  /**< Heading component of euler angles */
+    float roll;     /**< Roll component of euler angles */
+    float pitch;    /**< Pitch component of euler angles */
+} eul_t;
+
+/**
+ * @brief   Euclidian space representation structure
+ * To represent a position in three dimensional space.
+ */
+typedef struct {
+    float x;    /**< x component */
+    float y;    /**< y component */
+    float z;    /**< z component */
+} euclid_t;
+
+/**
+ * @name   The sensor default I2C address
+ * @{
+ */
+#define BNO055_ADDR_DEFAULT  (0x29U)
+/** @} */
+
+/**
+ * @brief   Status and error return codes
+ */
+enum {
+    BNO055_OK       = 0,      /**< exit without error */
+    BNO055_NOBUS    = ENXIO,  /**< cannot connect to module on i2c bus */
+    BNO055_NODEV    = ENODEV, /**< cannot read any data from module */
+    BNO055_NORW     = EIO,    /**< cannot read data from module */
+    BNO055_NOTREADY = EBUSY,  /**< no new data ready for reading */
+};
```

Comment:
This whole enum should be removed and the errno codes directly returned in the implementation. The doxygen documentation also needs to be adapted.